build.gradle

```diff
@@ -1832,12 +1840,17 @@ project(':streams') {
 
     // testCompileOnly prevents streams from exporting a dependency on test-utils, which would cause a dependency cycle
     testCompileOnly project(':streams:test-utils')
+    // KAFKA-14109
+    // The below compileOnly dependency is needed for JUnit 4 tests.
+    // It can be safely removed once all of streams has moved to JUnit 5.
+    testCompileOnly libs.junit4
+
```

Comment:
As we said below this is not needed if you move `libs.junitVintageEngine` back to `testImplementation`.

Reply:
Removed this dependency.